src/NServiceBus.Persistence.DynamoDB.PersistenceTests/OutboxTests.cs

```diff
@@ -204,4 +236,238 @@ public async Task Should_delete_operations_when_set_as_dispatched_and_more_opera
+        }
+
+        await ExpireMetadataRecord(incomingMessageId);
+
+        var result = await configuration.OutboxStorage.Get(incomingMessageId, configuration.GetContextBagForOutbox());
+        Assert.That(result, Is.Null);
+    }
+
+    [Test]
+    public async Task Should_return_fresh_entry_when_metadata_expired_but_phantom_record_overlap()
+    {
+        var incomingMessageId = Guid.NewGuid().ToString();
+        ContextBag fistAttemptContextBag = configuration.GetContextBagForOutbox();
+
+        var transportOperations = new TransportOperation[]
+        {
+            new(Guid.NewGuid().ToString(), new DispatchProperties(), ReadOnlyMemory<byte>.Empty, new Dictionary<string, string>()),
+            new(Guid.NewGuid().ToString(), new DispatchProperties(), ReadOnlyMemory<byte>.Empty, new Dictionary<string, string>()),
+            new(Guid.NewGuid().ToString(), new DispatchProperties(), ReadOnlyMemory<byte>.Empty, new Dictionary<string, string>()),
+            new(Guid.NewGuid().ToString(), new DispatchProperties(), ReadOnlyMemory<byte>.Empty, new Dictionary<string, string>())
+        };
+
+        using (var transaction = await configuration.OutboxStorage.BeginTransaction(fistAttemptContextBag))
+        {
+            var outboxMessage = new OutboxMessage(incomingMessageId, transportOperations);
+            await configuration.OutboxStorage.Store(outboxMessage, transaction, fistAttemptContextBag);
+
+            await transaction.Commit();
+        }
+
+        await ExpireMetadataRecord(incomingMessageId);
+
+        var transportOperationMessageId1 = Guid.NewGuid().ToString();
+        var transportOperationMessageId2 = Guid.NewGuid().ToString();
+        transportOperations = new TransportOperation[]
+        {
+            new(transportOperationMessageId1, new DispatchProperties(), ReadOnlyMemory<byte>.Empty, new Dictionary<string, string>()),
+            new(transportOperationMessageId2, new DispatchProperties(), ReadOnlyMemory<byte>.Empty, new Dictionary<string, string>()),
+        };
+
+        using (var transaction = await configuration.OutboxStorage.BeginTransaction(fistAttemptContextBag))
+        {
+            var outboxMessage = new OutboxMessage(incomingMessageId, transportOperations);
+            await configuration.OutboxStorage.Store(outboxMessage, transaction, fistAttemptContextBag);
+
+            await transaction.Commit();
+        }
+
+        var result = await configuration.OutboxStorage.Get(incomingMessageId, configuration.GetContextBagForOutbox());
+        Assert.AreEqual(transportOperations.Length, result.TransportOperations.Length);
+        Assert.That(result.TransportOperations[0].MessageId, Is.EqualTo(transportOperationMessageId1));
+        Assert.That(result.TransportOperations[1].MessageId, Is.EqualTo(transportOperationMessageId2));
+    }
+
+    [Test]
+    public async Task Should_return_fresh_entry_when_metadata_expired_but_phantom_record_overlap_beyond_query_size()
```

Comment:
from reading the test it isn't really clear what value this test provides over the previous one

Reply:
The thing is from an external behaviour perspective there is not much value because we can't assess what happens internally. I used this test to drive the implementation to make sure we do not query pages unnecessary.

With a unit test we could assert the further query is not executed but we'd have so simulate a ton on the mock